src/target/parser.rs

```diff
@@ -67,13 +67,19 @@ impl TargetInfoParserInner {
         let arch = cargo_env("CARGO_CFG_TARGET_ARCH", ft.map(|t| t.arch))?;
         let vendor = cargo_env("CARGO_CFG_TARGET_VENDOR", ft.map(|t| t.vendor))?;
         let os = cargo_env("CARGO_CFG_TARGET_OS", ft.map(|t| t.os))?;
-        let env = cargo_env("CARGO_CFG_TARGET_ENV", ft.map(|t| t.env))?;
+        let mut env = cargo_env("CARGO_CFG_TARGET_ENV", ft.map(|t| t.env))?;
         // `target_abi` was stabilized in Rust 1.78, which is higher than our
         // MSRV, so it may not always be available; In that case, fall back to
         // `""`, which is _probably_ correct for unknown target names.
-        let abi = cargo_env("CARGO_CFG_TARGET_ABI", ft.map(|t| t.abi))
+        let mut abi = cargo_env("CARGO_CFG_TARGET_ABI", ft.map(|t| t.abi))
             .unwrap_or_else(|_| String::default().into_boxed_str());
 
+        // Remove `macabi` and `sim` from `target_abi` (if present), it's been moved to `target_env`.
+        // TODO: Remove once MSRV is bumped to 1.91 and `rustc` removes these from `target_abi`.
+        if matches!(&*abi, "macabi" | "sim") {
+            env = mem::replace(&mut abi, String::default().into_boxed_str());
```

Comment:
do we want to have a debug assert that the previous value of `env` is either an empty string or same as `abi`?

```rust
debug_assert!(matches!(env, "" | "macbi" | "sim"), env);
```

Reply:
I've added a debug assertion now.